Plugins/SmithyCodeGenerator/SmithyCodeGeneratorPlugin.swift

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct Foundation.Data
+import class Foundation.FileManager
+import class Foundation.JSONDecoder
+import struct Foundation.URL
+import PackagePlugin
+
+@main
+struct SmithyCodeGeneratorPlugin: BuildToolPlugin {
```

Comment:
This is the build tool plugin implementation.
- Reads the `smithy-model-info.json` file to get the model location in the form of a file URL.  (All other source files in the target are ignored.)
- Constructs a build command that uses the `SmithyCodegenCLI` tool to read the model & generate a schemas file from it.